evaluate/report/csv.go

```diff
@@ -88,3 +68,59 @@ func (a AssessmentPerModel) Rows() (rows [][]string) {
 
 	return rows
 }
+
+// EvaluationFile holds the evaluation CSV file writer.
+type EvaluationFile struct {
+	// Holds the writer where the evaluation CSV is written to.
+	io.Writer
+}
+
+// NewEvaluationFile initializes an evaluation file and writes the corresponding CSV header.
+func NewEvaluationFile(writer io.Writer) (evaluationFile *EvaluationFile, err error) {
+	evaluationFile = &EvaluationFile{
+		Writer: writer,
+	}
+
+	var out strings.Builder
+	csv := csv.NewWriter(&out)
+
+	if err := csv.Write(evaluationHeader()); err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+	csv.Flush()
+
+	if _, err = evaluationFile.Writer.Write([]byte(out.String())); err != nil {
+		return nil, pkgerrors.WithStack(err)
+	}
+
+	return evaluationFile, nil
+}
+
+// WriteEvaluationRecord writes the assessments of a task into the evaluation CSV.
+func (e *EvaluationFile) WriteEvaluationRecord(model model.Model, language language.Language, repositoryName string, assessmentsPerTask map[task.Identifier]metrics.Assessments) (err error) {
+	var out strings.Builder
+	csv := csv.NewWriter(&out)
```

Comment:
Sorry for noticing just now. You should be able to avoid the `strings.Builder` and just do `csv.NewWriter(writer)`.